nengo/simulator.py

```diff
@@ -285,10 +291,18 @@ def run(self, time_in_seconds, progress_bar=None):
             For more control over the progress bar, pass in a `.ProgressBar`
             or `.ProgressUpdater` instance.
         """
-        steps = int(np.round(float(time_in_seconds) / self.dt))
-        logger.info("Running %s for %f seconds, or %d steps",
-                    self.model.label, time_in_seconds, steps)
-        self.run_steps(steps, progress_bar=progress_bar)
+        try:
+            steps = int(np.round(float(time_in_seconds) / self.dt))
+        except ZeroDivisionError:
+            steps = 0
```

Comment:
This can only produce a `ZeroDivisionError` when `dt == 0.`. In that case an exception should be raised because that is indeed an invalid value and running the simulator will never succeed. Though this might not be the place to check it (`Simulator.__init__` or `Model.__init__` might be better places and `ValueError` or `ValidationError` are probably better errors).

I could also see not checking `dt` at all because I don't think we have any guideline when to check argument ranges and when not? (We do it whereever the parameter system is involved by means of the assignment, but probably not for most other arguments.)

Reply:
Right... I did this because I figured the zero division happened here, but looking back at the original issue (#1354) the zero division happens in the progress bar when running for 0 timesteps. I guess I'll remove this, since now we only call `run_steps` if `steps > 0` but I guess I'll also fix the progress bar stuff...

In general I think we should always fail fast if we can.